### DISCUSSION TITLE
What is a Middleware and how to use it in ASP.NET Core?

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#48] [General]
# What is a Middleware and how to use it in ASP.NET Core?

Did you ever ask yourself: What is a middleware, and why should I use it?
If so, this blog post is exactly for you. We will see where we could use a middleware and also how we can use the Dependecy Injection container of ASP.NET Core.

https://steven-giesel.com/blogPost/16535182-1fb0-4fa9-8ffa-7b90e1355e5b?utm_source=csharpdigest&utm_medium=&utm_campaign=1526

**Comment by gadrochk** [comment]
We have IMiddleware nowadays (since 3.0). Apart from getting the signature right, it also requires DI registration which means it can be scoped. 

Side note. This comment required consent to “Know which resources you can access” and “Act on your behalf”. 
Really?!

**Reply by linkdotnet** [reply]
Good point. 

Regarding your side note: It was a trade-off to find a sound commenting system. A lot of people use Disqus but they have other issues. Giscus is completely open source, and it needs some way of creating an entry here.

How it works is that I have a discussion in a repository where giscus writes to. So that permission makes more or less sense - of course it could do that somewhere else as well. Anyway you can also navigate to the discussion thread on GitHub and directly write your comment (https://github.com/linkdotnet/Blog.Discussions/discussions). 
That is less convenient but doesn't need any permissions.

I don't have any plans to migrate to a different system, as this is more or less how it will work with all of those commenting systems (except I built this on my own). I do feel that the permissions are reasonable and the source is open.
